code/modules/antagonists/voidwalker/voidwalker_kidnap.dm

```diff
@@ -0,0 +1,173 @@
+/// Component that lets us space kidnap people as the voidwalker with our HAAAADS
+/datum/component/space_kidnap
+	/// How long does it take to kidnap them?
+	var/kidnap_time = 6 SECONDS
+	/// Are we kidnapping right now?
+	var/kidnapping = FALSE
+
+/datum/component/space_kidnap/Initialize(...)
+	if(!ishuman(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(parent, COMSIG_LIVING_UNARMED_ATTACK, PROC_REF(try_kidnap))
+
+/datum/component/space_kidnap/proc/try_kidnap(mob/living/parent, atom/target)
+	SIGNAL_HANDLER
+
+	if(!isliving(target))
+		return
+
+	var/mob/living/victim = target
+
+	if(victim.stat == DEAD)
+		target.balloon_alert(parent, "is dead!")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(!victim.incapacitated())
+		return COMPONENT_CANCEL_ATTACK_CHAIN
```

Comment:
We still want to cancel the attack chain here? This will shut off any other potential unarmed interaction at all